Content.Server/Terminator/Systems/TerminatorSystem.cs

```diff
@@ -63,4 +80,38 @@ public EntityUid CreateSpawner(EntityCoordinates coords, EntityUid? target)
 
         return uid;
     }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<TerminatorRoleComponent, MindComponent>();
+        while (query.MoveNext(out var uid, out var comp, out var mindComp))
+        {
+
+            if (comp.TerminationTime != null)
+            {
+                if (mindComp.OwnedEntity is not { } terminator)
+                    continue;
+
+                if (_timing.CurTime > comp.TerminationTime)
+                {
+                    _explosion.TriggerExplosive(terminator);
+                }
+
+                continue;
+            }
+
+            if (!_objectives.AllObjectivesComplete((uid, mindComp)))
+                continue;
+
+            if (mindComp.Session != null)
```

Comment:
Adding an expensive update check every tick is pretty ass, does adding a target objective component not work or smth.

Reply:
oh yeah this could be on the objective entity